browser/ui/webui/brave_wallet/android/swap_page_ui.cc

```diff
@@ -0,0 +1,162 @@
+#include "brave/browser/brave_wallet/swap_service_factory.h"
+#include "brave/browser/brave_wallet/tx_service_factory.h"
+
+#include "brave/components/brave_wallet/browser/blockchain_registry.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_constants.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+
+#include "brave/components/constants/webui_url_constants.h"
+
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/webui/sanitized_image_source.h"
+#include "chrome/browser/ui/webui/webui_util.h"
+#include "components/grit/brave_components_resources.h"
+#include "components/grit/brave_components_strings.h"
+#include "content/public/browser/url_data_source.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_ui.h"
+#include "content/public/browser/web_ui_data_source.h"
+#include "content/public/common/url_constants.h"
+
+#include "base/command_line.h"
+
+SwapPageUI::SwapPageUI(content::WebUI* web_ui, const std::string& name)
+    : ui::MojoWebUIController(web_ui,
+                              true /* Needed for webui browser tests */) {
+  content::WebUIDataSource* source =
+      content::WebUIDataSource::Create(kWalletPageHost);
+  web_ui->AddRequestableScheme(content::kChromeUIUntrustedScheme);
+
+  for (const auto& str : brave_wallet::kLocalizedStrings) {
+    std::u16string l10n_str =
+        brave_l10n::GetLocalizedResourceUTF16String(str.id);
+    source->AddString(str.name, l10n_str);
+  }
+
+  auto* profile = Profile::FromWebUI(web_ui);
+  webui::SetupWebUIDataSource(
+      source,
+      base::make_span(kBraveWalletSwapPageGenerated,
+                      kBraveWalletSwapPageGeneratedSize),
+      IDR_BRAVE_WALLET_SWAP_PAGE_HTML);
+
+  // Add required resources.
+  webui::SetupWebUIDataSource(
+      source,
+      base::make_span(kBraveWalletSwapPageGenerated,
+                      kBraveWalletSwapPageGeneratedSize),
+      IDR_BRAVE_WALLET_SWAP_PAGE_HTML);
+
+  source->AddString("braveWalletLedgerBridgeUrl", kUntrustedLedgerURL);
+  source->OverrideContentSecurityPolicy(
+      network::mojom::CSPDirectiveName::FrameSrc,
+      std::string("frame-src ") + kUntrustedTrezorURL + " " +
+          kUntrustedLedgerURL + " " + kUntrustedNftURL + " " +
+          kUntrustedMarketURL + ";");
+  source->AddString("braveWalletTrezorBridgeUrl", kUntrustedTrezorURL);
+  source->AddString("braveWalletNftBridgeUrl", kUntrustedNftURL);
+  source->AddString("braveWalletMarketUiBridgeUrl", kUntrustedMarketURL);
+  source->AddBoolean(brave_wallet::mojom::kP3ACountTestNetworksLoadTimeKey,
+                     base::CommandLine::ForCurrentProcess()->HasSwitch(
+                         brave_wallet::mojom::kP3ACountTestNetworksSwitch));
+  content::WebUIDataSource::Add(profile, source);
+
+//  content::URLDataSource::Add(profile,
+//                              std::make_unique<SanitizedImageSource>(profile));
```

Comment:
remove it please if we don't need it in the future.

Reply:
fixed